File: abstractions/dotnet/src/authentication/AnonymousAuthenticationProvider.cs
// ------------------------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
// ------------------------------------------------------------------------------

using System.Threading;
using System.Threading.Tasks;

namespace Microsoft.Kiota.Abstractions.Authentication
{
    /// <summary>
    /// This authentication provider does not perform any authentication.
    /// </summary>
    public class AnonymousAuthenticationProvider : IAuthenticationProvider
    {
        /// <summary>
        /// Authenticates the <see cref="RequestInformation"/> instance
        /// </summary>
        /// <param name="request">The request to authenticate</param>
        /// <param name="cancellationToken">The cancellation token for the task</param>
        /// <returns></returns>
        public Task AuthenticateRequestAsync(RequestInformation request, CancellationToken cancellationToken = default)
        {
            return Task.CompletedTask;
        }
    }
}

File: abstractions/php/src/Authentication/AnonymousAuthenticationProvider.php
<?php

namespace Microsoft\Kiota\Abstractions\Authentication;

use Http\Promise\FulfilledPromise;
use Http\Promise\Promise;
use Microsoft\Kiota\Abstractions\RequestInformation;

class AnonymousAuthenticationProvider implements AuthenticationProvider {

    /**
     * @param RequestInformation $request Request information
     * @return Promise
     */
    public function authenticateRequest(RequestInformation $request): Promise {
        return new FulfilledPromise(null);
    }
}
